### ISSUE TITLE
get_all_records() Taking 30-60 seconds

### ISSUE THREAD

**Issue by jasongurwin** [burnash/gspread#585]
I have a worksheet that is ~2,000 rows and 15 columns. When I run the get_all_records() command it takes anywhere from 30-60 seconds to run and sometimes never completes.

Is there anyway to speed this up or is this normal behavior?

Thanks...love the library.

**Comment by dalepo** [comment]
I'm facing the same issue, the function is not lazy or uses a generator. I think it would be good to make it generator-like function

**Comment by burnash** [comment]
Thank you for reporting this and for the feedback. Could you please share more details? What version of gspread are you using?

If you using gspread <3.0.0, please upgrade and check if the problem resolves. If you still experience the problem on the latest version, that's no good. Could you then try to retrieve the same data with `get_all_values()`, and if's still slow, with `spreadsheet.values_get(worksheet_title)`? Please report your findings here.

**Comment by aglotero** [comment]
Hey @burnash I'm facing the same issue, but the performance is inconsistent, sometimes took 4s to load all data, sometimes took 30-35 seconds. It seems to be a Google API problem rather than the library.

**Comment by aglotero** [comment]
@burnash @jasongurwin @dalepo my issue was solved:
the situation was: when a user is editing the spreadsheet and my service is trying to read the spreadsheet, I got a deadlock, my service needs to wait the user finish all editing in order to read the data. Whatever is the user is editing the Sheet A and I'm reading Sheet B, the lock is spreadsheet global.
In fact, my scenario was: on spreadsheet X user inputs some parameter on Sheet A, then call a user-defined function (google script) that calls my API, the API reads Sheet B from spreadsheet X, but the user started editing on Sheet A, so reading Sheet B is blocked.
Crazy :)
